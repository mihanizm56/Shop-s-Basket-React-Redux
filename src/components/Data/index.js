export default [
    {
        name:'Платье-миди с расклешенной юбкой',
        code:'82039-11',
        size:'44',
        color:'синий',
        id:'1-01'
    },
    {
        name:'Туфли женские украшенные кружевными вставками',
        code:'13524-01',
        size:'38',
        color:'чёрный',
        id:'1-02'
    },
    {
        name:'Платье-миди',
        code:'75039',
        size:'44',
        color:'белый',
        id:'1-03'
    },
    {
        name:'Платье-миди',
        code:'75039',
        size:'46',
        color:'белый',
        id:'1-04'
    }
]

