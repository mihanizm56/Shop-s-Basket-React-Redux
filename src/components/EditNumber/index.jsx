// @flow

import * as React from 'react';
import styled from 'styled-components';
// import PropTypes from 'prop-types';

const EditWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  width: 86px;
  height: 13px;
  margin-right: 174px;
`;

const Plus = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: 'Roboto-light';
  font-size: 22px;
  color: rgb(141, 141, 141);
  margin-right: 10px;
  line-height: 1.2;
  cursor: pointer;
`;

const NumberOfItems = styled.p`
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: 'Roboto';
  font-size: 16px;
  color: rgb(0, 0, 0);
  line-height: 1.2;
  text-align: center;
  margin: 0;
  padding: 0;
`;

const Minus = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: 'Roboto-light';
  font-size: 22px;
  color: rgb(141, 141, 141);
  margin-left: 10px;
  line-height: 1.2;
  cursor: pointer;
`;

// функции описать

type ObjectUser = {
  img: string,
  description: string,
  code: string,
  size: string,
  color: string,
  price: number,
  numberOfItems: number,
  id: number,
}

type TypedUserItems = Array<ObjectUser> | []

type typedProps = {
  userItems?: TypedUserItems,
  id?: ?number,
  numberOfItems?: ?number,
  increment?: any,
  decrement?: any,
};

export function EditNumber(props: typedProps) {
  const { userItems, id, numberOfItems, increment, decrement } = props;
  return (
    <EditWrapper>
      <Minus
        className="decrementFunc"
        onClick={() => decrement(userItems, id, numberOfItems)}
      >
        -
      </Minus>
      <NumberOfItems>{numberOfItems}</NumberOfItems>
      <Plus
        className="incrementFunc"
        onClick={() => increment(userItems, id, numberOfItems)}
      >
        +
      </Plus>
    </EditWrapper>
  );
}

EditNumber.defaultProps = {
  userItems: [],
  id: 0,
  numberOfItems: 0,
  increment: () => 'test_increment',
  decrement: () => 'test_decrement',
};

// EditNumber.propTypes = {
//   userItems: PropTypes.array.isRequired,
//   id: PropTypes.number.isRequired,
//   numberOfItems: PropTypes.number.isRequired,
//   increment: PropTypes.func.isRequired,
//   decrement: PropTypes.func.isRequired,
// }
