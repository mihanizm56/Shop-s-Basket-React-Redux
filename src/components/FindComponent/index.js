import React from "react"
import styled from "styled-components"
import zoom from "./img/zoom.png"

const FindWrapper = styled.div`
  display: flex;
  align-items: flex-end;
  width: 226px;
  height: 32px;
  border-bottom: 2px solid #2a2a2a;
`

const FindText = styled.input`
  margin-bottom: 2px;
  width: 203px;
  height: 20px;
  font-size: 15px;
  font-family: "Roboto";
  color: rgb(0, 0, 0);
  line-height: 1.2;
  border: none;
  outline: none;
  &::placeholder {
    font-size: 15px;
    font-family: "Roboto";
    color: rgb(0, 0, 0);
    line-height: 1.2;
  }
`

const ZoomIcon = styled.div`
  width: 19px;
  height: 19px;
  margin-bottom: 8px;
  background-image: url(${zoom});
  background-size: contain;
  background-repeat: no-repeat;
  cursor: pointer;
`

export const FindComponent = () => {
  return (
    <FindWrapper>
      <FindText
        className="findInput"
        type="text"
        placeholder="Поиск по сайту"
      />
      <ZoomIcon />
    </FindWrapper>
  )
}
