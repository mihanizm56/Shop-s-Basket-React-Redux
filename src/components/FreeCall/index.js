import React from "react";
import styled from "styled-components";

const FreeCallDiv = styled.div`
  font-size: 13px;
  font-family: "Roboto";
  margin-top: 2px;
  color: rgb(0, 0, 0);
  line-height: 1.2;
`;

export const FreeCall = () => (
    <FreeCallDiv>Бесплатный звонок по России</FreeCallDiv>
);
