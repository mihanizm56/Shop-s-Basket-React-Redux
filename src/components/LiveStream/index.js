import React from "react";
import styled from "styled-components";

const LiveStreamDiv = styled.div`
  font-size: 13px;
  font-family: "Roboto";
  color: rgb(0, 0, 0);
  line-height: 1.2;
`;

export const LiveStream = () => <LiveStreamDiv >Смотрите наш прямой эфир</LiveStreamDiv>