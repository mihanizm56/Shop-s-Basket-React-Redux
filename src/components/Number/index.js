import React from "react";
import styled from "styled-components"; 

const Bignumber = styled.span`
  font-size: 18px;
  font-family: "Roboto";
  color: rgb(0, 0, 0);
  line-height: 1.2;
  margin-right:8px;
`;

export const Number = () => <Bignumber>8 (800) 500-75-55</Bignumber>;
