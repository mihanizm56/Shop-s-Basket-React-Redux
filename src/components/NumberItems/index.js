import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const NumberItems = styled.p`
  width: 65px;
  margin: 0;
  padding: 0;
  font-size: 13px;
  font-family: "Roboto";
  color: rgb(0, 0, 0);
  line-height: 1.2;
`;

export const NumberOfItems = ({numberOfItems}) => {
  return (
    <NumberItems>
      В корзине: <b>{(numberOfItems > 4 || numberOfItems === 0) ? numberOfItems+' товаров' : numberOfItems+' товара'}</b>
    </NumberItems>
  );
};



NumberItems.defaultProps = {
  numberOfItems:0
};

NumberItems.propTypes = {
  numberOfItems:PropTypes.number.isRequired
};