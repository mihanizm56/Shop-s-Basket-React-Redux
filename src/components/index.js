export * from './EditNumber/'
export * from './FindComponent/'
export * from './FreeCall/'
export * from './LiveStream/'
export * from './Number/'
export * from './NumberItems/'
export * from './Star/'
export * from './YouTube/'
