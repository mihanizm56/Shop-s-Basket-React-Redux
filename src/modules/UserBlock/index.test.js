import React from "react"
import { shallow } from "enzyme"
import { UserBlock } from "./index"
import { NumberOfItems, FindComponent } from "../../components/"

describe("UserBlock test", () => {
  it("Renders correct without props", () => {
    const component = shallow(<UserBlock />)

    expect(component).toMatchSnapshot()
  })
  it("Renders correct with props", () => {
    const component = shallow(
      <UserBlock userName="test_name" numberOfItems={10} />
    )

    expect(component).toMatchSnapshot()
  })

  it("Renders correct NumberOfItems", () => {
    const component = shallow(<UserBlock />)
    const result = component.find(NumberOfItems).length

    expect(result).toBe(1)
  })
  it("Renders correct FindComponent", () => {
    const component = shallow(<UserBlock />)
    const result = component.find(FindComponent).length

    expect(result).toBe(1)
  })
})
