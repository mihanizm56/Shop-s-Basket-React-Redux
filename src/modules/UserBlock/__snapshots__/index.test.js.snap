// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock test Renders correct with props 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div />
      <FindComponent />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div />
        <styled.p>
          test_name
        </styled.p>
      </styled.div>
      <styled.div>
        <styled.div />
        <NumberOfItems
          numberOfItems={10}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`UserBlock test Renders correct without props 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div />
      <FindComponent />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div />
        <styled.p />
      </styled.div>
      <styled.div>
        <styled.div />
        <NumberOfItems
          numberOfItems={0}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
